riscv/src/compiler.rs

```diff
@@ -1346,6 +1431,54 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
             only_if_no_write_to_zero(format!("{rd} <== load_label({label});"), rd)
         }
 
+        // atomic and synchronization
+        "fence" | "fence.i" => vec![],
+
+        insn if insn.starts_with("lr.w") => {
```

Comment:
What are the options here?

Reply:
I think we should at least add an assertion about the alternatives, if that's possible.

Reply:
I did not find the exact syntax used by LLVM / GNU Binutils, but there are 4 alternatives: all possible states of bits aq and rl in the instruction encoding. Do you know where can we find it?